hathor/p2p/manager.py

```diff
@@ -132,7 +134,22 @@ def __init__(self, reactor: Reactor, my_peer: PeerId, server_factory: 'HathorSer
     def start(self) -> None:
         self.lc_reconnect.start(5, now=False)
         if settings.ENABLE_PEER_WHITELIST:
-            self.wl_reconnect.start(30)
+            self.start_whitelist_reconnect()
+
+    def start_whitelist_reconnect(self) -> None:
+        # The deferred returned by the LoopingCall start method
+        # executes when the looping call stops running
+        # https://docs.twistedmatrix.com/en/stable/api/twisted.internet.task.LoopingCall.html
+        d = self.wl_reconnect.start(30)
+        d.addErrback(self._handle_whitelist_reconnect_err)
+
+    def _handle_whitelist_reconnect_err(self, *args: Any, **kwargs: Any) -> None:
+        """ This method will be called when an exception happens inside the whitelist update
+            and ends up stopping the looping call.
+            We log the error and start the looping call again.
+        """
+        self.log.error('whitelist reconnect had an exception. Start looping call again.', args=args, kwargs=kwargs)
+        self.reactor.callLater(30, self.start_whitelist_reconnect)
```

Comment:
Why should we wait 30 seconds? Why not at once?

Reply:
The idea was to give some seconds for the server to recover because this errback might be called when an exception happens when getting the whitelist from the server. I used 30 just because it's the same interval between whitelist calls.

Reply:
Makes sense. It might be good to wait a bit to let the network recover.